test/test_ops.py

```diff
@@ -165,6 +169,20 @@ def unsupported(dtype):
 
         self.assertEqual(supported_backward_dtypes, claimed_backward_supported, msg=msg)
 
+    # Tests that the function and its (array-accepting) reference produce the same
+    #   values on the tensors from sample_inputs func for the corresponding op.
+    @onlyOnCPUAndCUDA
+    @suppress_warnings
+    @ops(ref_test_ops, allowed_dtypes=(torch.float32, torch.long))
+    def test_reference_testing(self, device, dtype, op):
+        # TODO: We should ideally use these tests for UnaryUfuncInfo as well, see: https://github.com/pytorch/pytorch/pull/59369/
+        if isinstance(op, (UnaryUfuncInfo, SpectralFuncInfo, ShapeFuncInfo)):
```

Comment:
I would just filter these out in the initial set (see note above).

Where are the SpectralFuncInfo and ShapeFuncInfo tests vs their references?

Reply:
For `ShapeFuncInfo` ops (there are only 2 funcs: `repeat` and `tile`): https://github.com/pytorch/pytorch/blob/5ec4ad7f54cc35ee40b5991bd65c974056048f84/test/test_shape_ops.py#L681

For `SpectralFuncInfo` ops: (all `fft` namespace ops):
https://github.com/pytorch/pytorch/blob/5ec4ad7f54cc35ee40b5991bd65c974056048f84/test/test_spectral_ops.py#L102

Reply:
Ah, OK. Only the 1D spectral ops are compared with their reference today, but it's fine to exclude them for now with a note. 

I think TestShapeFuncs can be deleted in favor of this new test, right? Doesn't it do the same thing?

Reply:
Agreed, `TestShapeFuncs` can be deleted - just that they test for multiple dtypes, I think just testing for `torch.float32, torch.long` is OK. Went through the documentation, didn't see anything specific to convince to test for complex types as well.

The dtypes the reference test for shape functions allow (currently) are here: https://github.com/pytorch/pytorch/blob/5ec4ad7f54cc35ee40b5991bd65c974056048f84/test/test_shape_ops.py#L679-L681

Just curious though, do we plan to add more ops in `ShapeFuncInfo` (which should not be reference tested) because there is an additional check in the test in `TestShapeFuncs`: https://github.com/pytorch/pytorch/blob/5ec4ad7f54cc35ee40b5991bd65c974056048f84/test/test_shape_ops.py#L680.

In the recent commit: I've added a note (for Spectral Funcs and Unary Ufuncs) and deleted `TestShapeFuncs` & added these funcs to new infra.

Reply:
We may add more in the future but let's not worry about it. As long as tile and repeat have their references tested by the new test we don't need to bother with the old test and it can be deleted.

Reply:
Great, thanks for the clarification @mruberry!